File: Java/Bink/src/net/qiujuer/blink/core/ReceiveDelivery.java
package net.qiujuer.blink.core;

import net.qiujuer.blink.listener.ReceiveListener;

/**
 * Receiver delivery interface
 */
public abstract class ReceiveDelivery {
    private final ReceiveListener mListener;

    public ReceiveDelivery(ReceiveListener listener) {
        mListener = listener;
    }


    /**
     * Get ReceiveListener
     *
     * @return ReceiveListener
     */
    protected ReceiveListener getReceiveListener() {
        return mListener;
    }

    /**
     * Parses a start response from the receiver.
     */
    public abstract void postReceiveStart(ReceivePacket entity);

    /**
     * Parses a end response from the receiver.
     */
    public abstract void postReceiveEnd(ReceivePacket entity, boolean isSuccess);

    /**
     * Parses a progress response from the receiver.
     */
    public abstract void postReceiveProgress(ReceivePacket entity, int total, int cur);
}

File: C Sharp/Blink/Blink/Listener/Delivery/ReceiveDelivery.cs
using Net.Qiujuer.Blink.Core;
using Net.Qiujuer.Blink.Listener;
using System;

namespace Net.Qiujuer.Blink.Listener.Delivery
{
    /// <summary>
    /// Receiver delivery interface
    /// </summary>
    public interface ReceiveDelivery : IDisposable
    {

        /**
         * Parses a start response from the receiver.
         */
        void PostReceiveStart(ReceivePacket entity);

        /**
         * Parses a end response from the receiver.
         */
        void PostReceiveEnd(ReceivePacket entity, bool isSuccess);

        /**
         * Parses a progress response from the receiver.
         */
        void PostReceiveProgress(ReceivePacket entity, float progress);

        /// <summary>
        /// On Socket Disconnect Post CallBack
        /// </summary>
        void PostBlinkDisconnect();
    }
}
